wasmtime-runtime/Cargo.toml

```diff
@@ -23,6 +23,13 @@ indexmap = "1.0.2"
 hashbrown = { version = "0.6.0", optional = true }
 spin = { version = "0.5.2", optional = true }
 thiserror = "1.0.4"
+goblin = "0.0.24"
+serde = { version = "1.0.99", features = ["derive"] }
+scroll = "0.9.2"
+gimli = "0.19.0"
+object = "0.12.0"
```

Comment:
Thanks for trimming the dependencies here. That said, wasmtime-runtime is still aiming to be a very light-weight core crate, so we don't want to have these kinds of dependencies, at least without an option to disable them.

Reply:
@sunfishcode Thanks, and I agree! There is another PR (yet submitted) I was working on for profiling wasmtime itself (along with jitted code) and I guarded that with a feature called profiling. I'll refactor this pull request for jitdump support to use that feature flag and resubmit soon. We can discuss from there any further improvements needed.

Reply:
@sunfishcode @yurydelendik  I refactored things a little bit after the rebase.

- Instead of having the jitdump support be a mod in runtime I made it a separate crate
- Added a feature "profiling" to conditionally compile the new "profiling" crate which holds the jitdump code. It is intended that this could hold ittapi code in a future commit
- Renamed some things from perf_* to jitdump_*.

One thing I wasn't sure how I wanted to handle was that the conditional compilation introduced the need to create a JitDumpAgent type on none profiling builds. I put this empty struct in wasmtime-runtime/lib.rs. There may be comment on this.

If there are any other comments let me know.